### PR TITLE
Add flake8 and imports check to pre-commit

### PR DESCRIPTION
**Pull request by naeioi**
This PR adds flake8 and imports check to pre-commit. 

They are previously checked in travis but NOT checked locally. As a result, contributors might devote extra effort to fix their code after finding out they are failing in travis' tests. 

**changlog**
- Remove `check-imports.sh` and `check-flake8.sh` scripts from `.travis.yml`.
- Check flake8 by `check-flake8.sh` and imports by pylint rule F401 in pre-commit. 
- Trigger pylint by pre-commit system hook. (bultin pre-commit hook [runs in isolated env](https://github.com/pre-commit/pre-commit-hooks/issues/157) and does not support dynamic import check). 
- Setup system-wide environment variables in `docker-entrypoint.sh` (previously in `check-import.sh` but now removed).

### CODE REVIEW

**Comment by jonashen** [comment]
Let's consider adding `check_imports.sh` as well.

**Comment by naeioi** [comment]
> Let's consider adding `check_imports.sh` as well.

Agree and done.

**Comment by naeioi** [comment]
@ryanjulian There are several reasons to use existing scripts rather than built-in support.
- Stronger consistancy
becuase we have single check entrypoint for local and CI.
- finer-grained control
e.g. for now we enforce docstring in code but not in tests. It's hard to control using only pre-commit built-ins.
- True dynamic import test
`check-imports.sh` will check imports by executing them at runtime. On the contrary flake8 runs statically. Built-in does not suffice if we were to have a consistent import check.

Plus, I think reusing code in travis for local test is not adding new tooling.

**Comment by ryanjulian** [comment]
* Fair. This distinction won't exist once we get flake8-compliant (and therefore get rid of the script), but fair.
* Fair. There will be no fine grained control needed if we're compliant and have a single `setup.cfg`
* pylint does actual tests of imports. If you turn on F401 in `setup.cfg`I think you will get the same effect as `check_imports.sh`. I didn't know about this feature at the time I wrote it.

**Comment by ryanjulian** [comment]
Can you check if we can turn on F401 in pylint and get rid of check_imports.sh? The local hook for flake8 is okay until we can clear out all the flake8 bugs.

**Comment by ryanjulian** [comment]
Doesn't this also mean you can remove the hooks to `check_flake8.sh` and `check_imports.sh` from `.travis.yml`?

**Comment by naeioi** [comment]
> Doesn't this also mean you can remove the hooks to `check_flake8.sh` and `check_imports.sh` from `.travis.yml`?

I think so - As travis runs pre-commit.

**Comment by naeioi** [comment]
>     * pylint does actual tests of imports. If you turn on F401 in `setup.cfg`I think you will get the same effect as `check_imports.sh`. I didn't know about this feature at the time I wrote it.
Are you sure about this? I checked and found out that F401 in enabled in check_flake. If flake8 actually executes the import, it should fail in CI when testing import of AntEnv, because mujoco library is not yet set up in `LD_LIBRARY_PATH` - it is set in `check_imports.sh` which runs after `check_flake8.sh`.

**Comment by ryanjulian** [comment]
I'm fairly sure about pylint. I don't think flake8 does this check. flake8 and pylint are separate tools.

**Comment by naeioi** [comment]
flake8 is checked by `check_flake8.sh` and import checked by pylint now. Both are configured in pre-commit. Redundancy calls to check script in `.travis.yml` have been removed.

I will remove intentionally blocking tests once approved.

File: scripts/travisci/check_precommit.sh [hunk 1]
```diff
@@ -1,4 +1,9 @@
 #!/usr/bin/env bash
+export PYTHONPATH=.
```
Comment (ryanjulian) [inline scripts/travisci/check_precommit.sh]: isn't this done in the docker entrypoint?

Comment (naeioi) [reply scripts/travisci/check_precommit.sh]: It 's not set elsewhere but i found `conda activate garage` does the thing so we are safe to remove it.

**Comment by naeioi** [comment]
Would like have your reviews @gonzaiva @jonashen

**Comment by naeioi** [comment]
I fear that we need to revert to use custom script to check import because pre-commit's built-in pylint runs in an isolated virtualenv.
https://github.com/pre-commit/pre-commit/issues/178

**Comment by ryanjulian** [comment]
The `system` workaround seems fine to me, if it works. If not, we can just run pylint outside of pre-commit.

Let me know what you want to do, preferably in the form of a PR :P

**Comment by naeioi** [comment]
Resolve dynamic import check in https://github.com/rlworkgroup/garage/pull/422/commits/ec76295e973f74297e801b9fb33221db9359f27b. PR is overkill IMO. I will apply this fix to akro similarly.

File: scripts/travisci/check_precommit.sh [hunk 2]
```diff
@@ -1,4 +1,8 @@
 #!/usr/bin/env bash
+export TF_CPP_MIN_LOG_LEVEL=3      # shut TensorFlow up
```
[the remark this answers is not in the document]
Comment (naeioi) [reply scripts/travisci/check_precommit.sh]: You mean `conda activate garage` does the job? The only occurrence of `TF_CPP_MIN_LOG_LEVEL` is here in `check_precommit.sh`

[the remark this answers is not in the document]
Comment (ryanjulian) [reply scripts/travisci/check_precommit.sh]: i was referring to `docker/docker_entrypoint.sh` which should do all of these things before running the code.

[the remark this answers is not in the document]
Comment (naeioi) [reply scripts/travisci/check_precommit.sh]: i agree. These exports were originally improperly placed in `check_imports.sh` but not in entrypoint. I will move them there.

**Pull request merged**